jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcTenantMigrationRepository.java

```diff
@@ -0,0 +1,77 @@
+package io.kestra.jdbc.repository;
+
+import io.kestra.core.repositories.TenantMigrationInterface;
+import io.kestra.jdbc.JooqDSLContextWrapper;
+import java.util.List;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+
+public abstract class AbstractJdbcTenantMigrationRepository implements TenantMigrationInterface {
+
+    protected final JooqDSLContextWrapper dslContextWrapper;
+
+    protected AbstractJdbcTenantMigrationRepository(JooqDSLContextWrapper dslContextWrapper) {
+        this.dslContextWrapper = dslContextWrapper;
+    }
+
+    public void migrateTenant(boolean dryRun) {
+        migrate(dryRun);
+    }
+
+    public void migrate(boolean dryRun) {
+        List<Table<?>> tables = dslContextWrapper.transactionResult(configuration -> {
+            DSLContext context = DSL.using(configuration);
+            return context.meta().getTables();
+        });
+
+        System.out.printf("📦 Found %d tables.\n\n", tables.size());
+
+        int totalAffected = 0;
+
+        for (Table<?> table : tables) {
+            Field<String> tenantField = table.field("tenant_id", String.class);
+
+            if (tenantField == null) {
+                continue;
+            }
+
+            Condition condition = tenantField.isNull();
+
+            int count = dslContextWrapper.transactionResult(configuration -> {
+                DSLContext context = DSL.using(configuration);
+                return context.selectCount()
+                    .from(table)
+                    .where(condition)
+                    .fetchOne(0, int.class);
+            });
```

Comment:
If it's only for reporting, I would not do that as for huge table it can be very costly.
Reporting updated is enought

Reply:
In fact, reading the code , you should do that only for dryRun